### ISSUE TITLE
[Bug]:  Health-check fails with working endpoints

### ISSUE THREAD

**Issue by lalineanegra** [BerriAI/litellm#3987]
### What happened?

When using the health check endpoint to verify the endpoints in the config.yaml, many of them fail, although they work in reality

### Relevant log output

```shell
Pre-API Call
14:57:44 - LiteLLM:INFO: utils.py:1307 - 
Request Sent from LiteLLM:


Pre-API Call
Post-API Call
Traceback (most recent call last):
  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/api_core/grpc_helpers_async.py", line 85, in __await__
    response = yield from self._call.__await__()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/imagemaker/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/grpc/aio/_call.py", line 318, in __await__
    raise _create_rpc_error(
grpc.aio._call.AioRpcError: <AioRpcError of RPC that terminated with:
        status = StatusCode.INVALID_ARGUMENT
        details = "Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001."
        debug_error_string = "UNKNOWN:Error received from peer ipv4:172.xxx.xxx.xx:443 {created_time:"2024-06-03T14:57:45.435105-04:00", grpc_status:3, grpc_message:"Invalid Endpoint name: projects/my-porject-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001."}"
>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/imagemaker/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/llms/vertex_ai.py", line 1164, in async_completion
    response_obj = await llm_model.predict(
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/prediction_service/async_client.py", line 412, in predict
    response = await rpc(
               ^^^^^^^^^^
  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/api_core/grpc_helpers_async.py", line 88, in __await__
    raise exceptions.from_grpc_error(rpc_error) from rpc_error
google.api_core.exceptions.InvalidArgument: 400 Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/main.py", line 347, in acompletion
    response = await init_response
               ^^^^^^^^^^^^^^^^^^^
  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/llms/vertex_ai.py", line 1241, in async_completion
    raise VertexAIError(status_code=500, message=str(e))
litellm.llms.vertex_ai.VertexAIError: 400 Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.
On Async Failure !

kwargs {'model': 'textembedding-gecko-multilingual@001', 'messages': [{'role': 'user', 'content': "Hey how's it going?"}], 'optional_params': {}, 'litellm_params': {'acompletion': True, 'api_key': None, 'force_timeout': 600, 'logger_fn': None, 'verbose': False, 'custom_llm_provider': 'vertex_ai', 'api_base': '', 'litellm_call_id': '9f33bf76-8990-4405-9765-204412cd96b1', 'model_alias_map': {}, 'completion_call_id': None, 'metadata': None, 'model_info': None, 'proxy_server_request': None, 'preset_cache_key': None, 'no-log': False, 'stream_response': {}, 'input_cost_per_token': None, 'input_cost_per_second': None, 'output_cost_per_token': None, 'output_cost_per_second': None}, 'start_time': datetime.datetime(2024, 6, 3, 14, 57, 41, 21705), 'stream': False, 'user': None, 'call_type': 'acompletion', 'litellm_call_id': '9f33bf76-8990-4405-9765-204412cd96b1', 'completion_start_time': None, 'vertex_project': 'ai-lab-plygrd-9d5c', 'vertex_location': 'us-central1', 'custom_llm_provider': 'vertex_ai', 'input': "Hey how's it going?", 'api_key': None, 'additional_args': {'complete_input_dict': {}, 'request_str': ''}, 'log_event_type': 'failed_api_call', 'exception': APIError('VertexAIException APIError - 400 Invalid Endpoint name: projects/ai-lab-plygrd-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.'), 'traceback_exception': 'Traceback (most recent call last):\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/api_core/grpc_helpers_async.py", line 85, in __await__\n    response = yield from self._call.__await__()\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/grpc/aio/_call.py", line 318, in __await__\n    raise _create_rpc_error(\ngrpc.aio._call.AioRpcError: <AioRpcError of RPC that terminated with:\n\tstatus = StatusCode.INVALID_ARGUMENT\n\tdetails = "Invalid Endpoint name: projects/ai-lab-plygrd-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001."\n\tdebug_error_string = "UNKNOWN:Error received from peer ipv4:172.217.192.95:443 {created_time:"2024-06-03T14:57:45.435105-04:00", grpc_status:3, grpc_message:"Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001."}"\n>\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/llms/vertex_ai.py", line 1164, in async_completion\n    response_obj = await llm_model.predict(\n                   ^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/cloud/aiplatform_v1/services/prediction_service/async_client.py", line 412, in predict\n    response = await rpc(\n               ^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/google/api_core/grpc_helpers_async.py", line 88, in __await__\n    raise exceptions.from_grpc_error(rpc_error) from rpc_error\ngoogle.api_core.exceptions.InvalidArgument: 400 Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/main.py", line 347, in acompletion\n    response = await init_response\n               ^^^^^^^^^^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/llms/vertex_ai.py", line 1241, in async_completion\n    raise VertexAIError(status_code=500, message=str(e))\nlitellm.llms.vertex_ai.VertexAIError: 400 Invalid Endpoint name: projects/my-project-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/utils.py", line 3753, in wrapper_async\n    result = await original_function(*args, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/main.py", line 369, in acompletion\n    raise exception_type(\n          ^^^^^^^^^^^^^^^\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/utils.py", line 9981, in exception_type\n    raise e\n  File "/Users/me/work/latam/litellm-proxy/.venv/lib/python3.11/site-packages/litellm/utils.py", line 9185, in exception_type\n    raise APIError(\nlitellm.exceptions.APIError: VertexAIException APIError - 400 Invalid Endpoint name: projects/ai-lab-plygrd-9d5c/locations/us-central1/endpoints/textembedding-gecko-multilingual@001.\n', 'end_time': datetime.datetime(2024, 6, 3, 14, 57, 45, 456250), 'original_response': None}
Exception: Model is None and does not exist in passed completion_response. Passed completion_response=None, model=None
14:57:45 - LiteLLM Proxy:INFO: parallel_request_limiter.py:500 - Inside Parallel Request Limiter: An exception occurred - 'NoneType' object has no attri
```

**Comment by ghost** [comment]
hey @lalineanegra just set the `mode` in your config, to say what kind of model this is - in this case it would be 

```yaml
model_list:
  - model_name: azure-embedding-model
    litellm_params:
      model: azure/azure-embedding-model
      api_base: os.environ/AZURE_API_BASE
      api_key: os.environ/AZURE_API_KEY
      api_version: "2023-07-01-preview"
    model_info:
      mode: embedding # 👈 ADD THIS
```

https://docs.litellm.ai/docs/proxy/health#embedding-models

i also pushed a fix to auto infer this - https://github.com/BerriAI/litellm/commit/90f5aa7125ed5617a4334c6eab14af0687f1361b

**Comment by lalineanegra** [comment]
> hey @lalineanegra just set the `mode` in your config, to say what kind of model this is - in this case it would be
> 
> ```yaml
> model_list:
>   - model_name: azure-embedding-model
>     litellm_params:
>       model: azure/azure-embedding-model
>       api_base: os.environ/AZURE_API_BASE
>       api_key: os.environ/AZURE_API_KEY
>       api_version: "2023-07-01-preview"
>     model_info:
>       mode: embedding # 👈 ADD THIS
> ```
> 
> https://docs.litellm.ai/docs/proxy/health#embedding-models
> 
> i also pushed a fix to auto infer this - [90f5aa7](https://github.com/BerriAI/litellm/commit/90f5aa7125ed5617a4334c6eab14af0687f1361b)

And in the case of text to speech models, nothing needs to be added in model_info?

**Comment by ghost** [comment]
i think that's not covered by `/health` yet - can you share your tts config i'll add support for it
